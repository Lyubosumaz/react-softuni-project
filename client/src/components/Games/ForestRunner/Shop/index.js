import { PropTypes } from 'prop-types';
import { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import ItemsList from '../../../../components/ItemsList';
import { setNotification } from '../../../../components/Notification/actions';
import { httpGame } from '../../../../services/http';

function Shop({ setNotificationError }) {
    const [items, setItems] = useState([]);

    useEffect(() => {
        httpGame
            .shop()
            .then((i) => {
                if (!i) {
                    return;
                }

                setItems(i);
            })
            .catch((err) => {
                setNotificationError(err);
            });
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return (
        <div>
            <h1>Shop</h1>

            <ItemsList items={items} />
        </div>
    );
}

function mapDispatchToProps(dispatch) {
    return {
        setNotificationError: (data) => dispatch(setNotification(data).error()),
    };
}

export default connect(null, mapDispatchToProps)(Shop);

Shop.propTypes = {
    setNotificationError: PropTypes.func.isRequired,
};
