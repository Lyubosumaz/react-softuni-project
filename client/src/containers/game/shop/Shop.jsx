import { useState, useEffect } from 'react';
import { httpGame } from '../../../services/http';
import { toast } from 'react-toastify';
import ItemCard from '../../../components/ItemCard';
import { toastError } from '../../../utils/toastHandler';

export default function Shop() {
    const [items, setItems] = useState([]);

    useEffect(() => {
        httpGame
            .shop()
            .then((i) => {
                if (!i) {
                    return;
                }

                const rows = [...Array(Math.ceil(i.length / 4))];
                const arr = rows.map((row, index) => {
                    return i.slice(index * 4, index * 4 + 4);
                });

                setItems(arr);
            })
            .catch((err) => {
                toastError(err);
            });
    }, []);

    return (
        <div>
            <h1>Shop</h1>

            <div className="item-card-container">
                {items &&
                    items.map((row, i) => {
                        return (
                            <div key={i} className="item-card-card">
                                {row.map((item, index) => {
                                    return <ItemCard key={index} item={item} />;
                                })}
                            </div>
                        );
                    })}
            </div>
        </div>
    );
}
