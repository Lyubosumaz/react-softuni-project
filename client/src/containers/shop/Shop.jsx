import React from 'react';

export default function Shop() {
    return (
        <div>
            <div>Hello Shop</div>
        </div>
    );
};