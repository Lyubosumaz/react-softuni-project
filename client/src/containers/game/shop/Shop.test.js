import React from 'react';
import { shallow } from 'enzyme';
import Shop from './Shop';

it('Renter Shop component', () => {
    const wrapper = shallow(<Shop />);
    expect(wrapper).toBeTruthy();
});

it('Checks for html changes', () => {
    const wrapper = shallow(<Shop />);
    expect(wrapper).toMatchSnapshot();
});
