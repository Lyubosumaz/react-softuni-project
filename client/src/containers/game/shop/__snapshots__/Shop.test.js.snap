// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks for html changes 1`] = `
<div>
  <h1>
    Shop
  </h1>
  <div
    className="item-card-container"
  />
</div>
`;
